java/src/main/java/ai/rapids/cudf/DecimalUtils.java

```diff
@@ -82,13 +82,15 @@ public static ColumnVector lessThan(ColumnView lhs, BigDecimal rhs) {
     int leftScale = lhs.getType().getScale();
     int leftPrecision = lhs.getType().getDecimalMaxPrecision();
 
-    // First we have to round the scalar (rhs) to the same scale as lhs.  Because this is a
-    // less than and it is rhs that we are rounding, we will round away from 0 (UP)
-    // to make sure we always return the correct value.
+    // First we have to round the scalar (rhs) to the same scale as lhs.
+    // Because this is a less-than, and it is rhs that we are rounding, we will round away from 0 (UP) in case rhs is
+    // positive and toward 0 (DOWN) if rhs is negative to make sure we always return the correct value.
```

Comment:
A simpler way to describe this is that we want to round towards positive infinity, i.e.: prefer to move to the next most positive value when throwing away precision.